larky/src/test/resources/stdlib_tests/xml/etree/test_elementtree.star

```diff
@@ -87,12 +88,36 @@ def _test_xpath():
     asserts.eq(1, len(root.findall("./actress[1]")))
     asserts.eq('1984', root.findall("./actor[2]/born")[0].text)
     asserts.eq('8', root.findall("./actor/film[1]")[0].text)
+    # test search by attrib
+    asserts.eq('3', root.findall("./actor/*[@updated='Yes']")[0].text)
+    asserts.eq('actress', root.findall(".//*[@name='Jenny']")[0].tag)
+
+
+def _test_update_and_serialize():
+    parser = SimpleXMLTreeBuilder.TreeBuilder()
+    data = ''.join(['<data><teacher name="Jenny"><born>1983</born></teacher>',
+    '<student name="Tim"><performance><Grade>A+</Grade></performance><info><born>2005</born></info></student>',
+    '<student name="John"><performance><Grade>B</Grade></performance><info><born>2004</born></info></student></data>'])
+
+    tree = parse(data, parser)
+    root = tree.getroot()
+
+    # test update node text
+    root.findall(".//*[@name='John']/performance/Grade")[0].text = 'A-'
```

Comment:
holy shit, this works?

Reply:
yes, works on fields like attrib, text, tag in Element mutablestruct